orttraining/orttraining/core/graph/gradient_builder_base.h

```diff
@@ -217,7 +217,7 @@ class GradientBuilderBase {
     if (!name.empty()) {
       unique_prefix << name << "_Grad/";
     } else {
-      unique_prefix << node_->OpType() << "_" << node_->Index() << "_Grad/";
+      unique_prefix << node_->OpType() << "_index_" << node_->Index() << "_Grad/";
```

Comment:
To eliminate the future coincidence, let's ask graph to generate the name for us 
graph_->GenerateNodeName(node_->OpType())